python/lsst/summit/utils/guiders/reading.py

```diff
@@ -763,30 +774,87 @@ def getGuiderRawStamps(self, dayObs: int, seqNum: int) -> dict[str, Stamps]:
+        The mask is built on the bias-subtracted data to detect only residual
+        bad columns (e.g., bright saturated star trails) that were NOT removed
+        by the bias subtraction. Original stamps are never modified.
 
         Parameters
         ----------
         rawStampsDict : `dict[str, Stamps]`
-            ROI view of stamps that come from butler `guider_raw`.
+            ROI view of stamps from butler `guider_raw`.
+        doSubtractMedian : `bool`, optional
+            If True, subtract median row bias.
+        columnMaskK : `float`, optional
+            Threshold factor for column mask detection (default 50.0).
+            Higher values mask only brighter outlier columns.
+
+        Returns
+        -------
+        processedStamps : `dict[str, Stamps]`
+            New Stamps objects with processed (copied) data.
+            Per-stamp column masks are included in each stamp's MaskedImageF.
         """
+        processedStamps: dict[str, Stamps] = {}
+
         for detName, stamps in rawStampsDict.items():
-            # Subtract median along the row axis
-            rowAxis = 0
+            if len(stamps) == 0:
+                processedStamps[detName] = stamps
+                continue
+
+            stampList: list[Stamp] = []
             for i in range(len(stamps)):
-                # Get the stamp image array
-                data = stamps[i].stamp_im.image.array
-                # Subtract median along the specified row axis
-                medianRow = np.nanmedian(data, axis=rowAxis, keepdims=True)
-                # Update the stamp with the new data
-                stamps[i].stamp_im.image.array = data - medianRow
-            # Update the rawStampsMap in-place
-            rawStampsDict[detName] = stamps
-        return rawStampsDict
+                # Work on a copy - never modify original
+                data = stamps[i].stamp_im.image.array.copy()
+
+                # 0. Compute median row bias
+                medianRows = np.nanmedian(data, axis=0)
+                medianValue = np.nanmedian(data)
+                # Replace NaN medians (fully masked columns) with global median
+                medianRows = np.where(np.isnan(medianRows), medianValue, medianRows)
+
+                # 1. Apply column mask FIRST - set masked pixels to NaN
+                colMask = getColumnMask(data - medianRows[np.newaxis, :], k=columnMaskK)
+
+                # Create mask array for MaskedImageF (BAD=1 for masked cols)
+                nRows, nCols = data.shape
+                maskArray = np.zeros((nRows, nCols), dtype=np.uint32)
+                maskArray[colMask] = 1  # Set BAD bit for masked columns
+
+                if doSubtractMedian:
+                    data = data - medianRows[np.newaxis, :]
+
+                # 3. Fill masked columns with global median
+                if colMask.any():
+                    medianValue = np.nanmedian(data[~colMask]) if (~colMask).any() else 0.0
```

Comment:
If you can use `np.nan` instead of `0.0` it might be better, but don't worry too much if that causes problems, as I think if that's the case then everything is totally broken if this is the case anyway.